### PR TITLE
UI for mock googlepay token creation and conversion

### PR DESCRIPTION
**Pull request by elizalucas**
This is a PR for the UI to generate and convert a google pay token in payments sample app. 

- In smokebox/sandbox we will autogenerate a mock googlepay token (implemented in this PR)
- In stg/prod we will implement the real google pay button which will call the Google Pay API and return a real googlepay token (not implemented in this PR)
- The generated tokens will populate the form fields (implemented in this PR for the mock token)
- The convert token will call the endpoint in wallets-api and return the converted token (not implemented in this PR

Screen Shot 2022-03-01 at 12 30 09 PM
)

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -5602,7 +5607,7 @@ expect@^27.4.2:
     jest-message-util "^27.4.2"
     jest-regex-util "^27.4.0"
 
-express@^4.17.3:
+express@4.17.3:
```
Comment (timmy-circle) [inline yarn.lock:5610]: is it intentional?

Comment (elizalucas) [reply yarn.lock]: nope, not sure why that's in there.. just pushed an additional commit to remove

File: pages/debug/payments/googlepay/convertToken.vue [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        <v-form>
+          <v-btn
+            depressed
+            class="mb-7"
+            color="primary"
+            :loading="loading"
+            @click.prevent="autogenerateToken()"
+          >
+            Autogenerate token
+          </v-btn>
```
Comment (timmy-circle) [inline pages/debug/payments/googlepay/convertToken.vue:14]: Should this only be shown in sandbox/smokebox?

File: pages/debug/payments/googlepay/convertToken.vue [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  makeApiCall() {
+    const type = this.formData.type === 'Google Pay' ? 'googlepay' : 'applepay'
+    let payload = null
+
+    switch (type) {
+      case 'googlepay':
+        payload = {
+          type,
+          token_data: {
+            protocolVersion: 'ECv1',
```
Comment (timmy-circle) [inline pages/debug/payments/googlepay/convertToken.vue:106]: should this be a customer inputed value as well? If there's only 1 possible value. Let's do a drop down with only one selection pre-selected?

File: pages/debug/payments/googlepay/convertToken.vue [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        }
+        break
+      case 'applepay':
+        payload = {
+          type,
+          token_data: 'TODO: implement apply pay payload',
+        }
+    }
+
+    console.log(payload)
```
Comment (timmy-circle) [inline pages/debug/payments/googlepay/convertToken.vue:119]: should we be logging this?

Comment (elizalucas) [reply pages/debug/payments/googlepay/convertToken.vue]: removed this line

File: pages/debug/payments/googlepay/convertToken.vue (hunk 2 above)
Comment (elizalucas) [reply pages/debug/payments/googlepay/convertToken.vue]: Implemented conditional in latest commit

**Comment by elizalucas** [comment]
> lgtm, but I wonder if we could add some unit test

Happy to add unit tests - are you thinking a test for the token conversion? In that case I could add the test in when I actually wire up the button to the endpoint

**Review by timmy-circle — APPROVED** [review]
lgtm. Just one nit.

**Comment by huaweigu** [comment]
> > lgtm, but I wonder if we could add some unit test
> 
> Happy to add unit tests - are you thinking a test for the token conversion? In that case I could add the test in when I actually wire up the button to the endpoint

yes, thanks

File: yarn.lock (hunk 1 above)
Comment (antiv0) [reply yarn.lock]: (that happens with yarn add dependency to me also and then you need to go manually remove it when that happens)

File: package.json [hunk 5]
```diff
@@ -27,6 +27,7 @@
     "@nuxtjs/dotenv": "1.4.1",
     "@nuxtjs/vuetify": "1.12.2",
     "@types/card-validator": "7.0.1",
+    "@types/googlepay": "^0.6.4",
```
Comment (antiv0) [inline package.json:30]: you shouldn't have these `^` in here

**Pull request merged**